pkg/engine/executor/executor_test.go

```diff
@@ -283,3 +283,230 @@ func TestNewParsePipeline(t *testing.T) {
+			},
+		},
+		{
+			name: "accept JSON numbers as strings (v1 compatibility)",
+			schema: arrow.NewSchema([]arrow.Field{
+				{Name: types.ColumnNameBuiltinMessage, Type: arrow.BinaryTypes.String},
+			}, nil),
+			input: arrowtest.Rows{
+				{"message": `{"status": 200, "port": 8080, "timeout": 30.5, "retries": 0}`},
+				{"message": `{"level": "info", "pid": 12345, "memory": 256.8}`},
+				{"message": `{"score": -1, "version": 2.1, "enabled": true}`},
+			},
+			requestedKeys:  []string{"status", "port", "timeout", "pid", "memory", "score", "version"},
+			expectedFields: 8, // message, memory, pid, port, score, status, timeout, version
+			expectedOutput: arrowtest.Rows{
+				{"message": `{"status": 200, "port": 8080, "timeout": 30.5, "retries": 0}`, "memory": nil, "pid": nil, "port": "8080", "score": nil, "status": "200", "timeout": "30.5", "version": nil},
+				{"message": `{"level": "info", "pid": 12345, "memory": 256.8}`, "memory": "256.8", "pid": "12345", "port": nil, "score": nil, "status": nil, "timeout": nil, "version": nil},
+				{"message": `{"score": -1, "version": 2.1, "enabled": true}`, "memory": nil, "pid": nil, "port": nil, "score": "-1", "status": nil, "timeout": nil, "version": "2.1"},
+			},
+		},
+		{
+			name: "mixed nested objects and numbers",
+			schema: arrow.NewSchema([]arrow.Field{
+				{Name: types.ColumnNameBuiltinMessage, Type: arrow.BinaryTypes.String},
+			}, nil),
+			input: arrowtest.Rows{
+				{"message": `{"request": {"url": "/api/users", "method": "GET"}, "response": {"status": 200, "time": 45.2}}`},
+				{"message": `{"user": {"id": 123, "profile": {"age": 25}}, "active": true}`},
+			},
+			requestedKeys:  nil, // Extract all keys
+			expectedFields: 8,   // message, active, request_method, request_url, response_status, response_time, user_id, user_profile_age
+			expectedOutput: arrowtest.Rows{
+				{"message": `{"request": {"url": "/api/users", "method": "GET"}, "response": {"status": 200, "time": 45.2}}`, "active": nil, "request_method": "GET", "request_url": "/api/users", "response_status": "200", "response_time": "45.2", "user_id": nil, "user_profile_age": nil},
+				{"message": `{"user": {"id": 123, "profile": {"age": 25}}, "active": true}`, "active": "true", "request_method": nil, "request_url": nil, "response_status": nil, "response_time": nil, "user_id": "123", "user_profile_age": "25"},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			alloc := memory.NewCheckedAllocator(memory.DefaultAllocator)
+			defer alloc.AssertSize(t, 0) // Assert empty on test exit
+
+			// Create input data with message column containing JSON
+			input := NewArrowtestPipeline(
+				alloc,
+				tt.schema,
+				tt.input,
+			)
+
+			// Create ParseNode for JSON parsing
+			parseNode := &physical.ParseNode{
+				Kind:          physical.ParserJSON,
+				RequestedKeys: tt.requestedKeys,
+			}
+
+			pipeline := NewParsePipeline(parseNode, input, alloc)
```

Comment:
Since this is only testing NewParsePipeline and not the whole executor, should this be moved to a `parse_test.go` to align it more closely with its implementation? That might make the tests easier to discover when browsing files. 

(Ditto with the test above for logfmt; I think I missed how the scope of the test changed as we evolved the last PR)